charts/spire/charts/spire-server/templates/statefulset.yaml

```diff
@@ -125,6 +125,11 @@ spec:
               mountPath: /run/spire/upstream_ca
               readOnly: false
             {{ end }}
+            {{- if eq (.Values.upstreamAuthority.spire.enabled | toString) "true" }}
```

Comment:
This would fail some conventions that people might use to say that something is not enabled.

For example, as a boolean value, zero values evaulate to false, and non-zero values evaluate to true.  This means that

| false, 0, ""

evaluate to false while

| true, (any number), (any non-empty string)

evaluates to true.

This means that our "check" for true is stronger than the checks the template package that Helm is built on top of does.  This could lead to the unexpected behavior of

| upstreamAuthority.spire.enabled = "True" 

failing because we are doing a string comparison, and "True" != "true" yields the Boolean false.  When if we didn't do string comparison, the logic would be "True" is non-zero, yielding the Boolean value true

Strongly consider using Boolean evaluation instead of string matching.

Reply:
Overcome by differences in values now. Done.